Makefile

```diff
@@ -3,50 +3,13 @@ all: ci rhel
 
 # The CI system uses the OpenShift CI public key and verifies it against a bucket on GCS.
 ci:
-	keydir=$(shell mktemp -d -t keys-XXXXXXXX); \
-	gpg --dearmor < keys/verifier-public-key-openshift-ci > "$$keydir/verifier-public-key-ci.gpg"; \
-	gpg --dearmor < keys/verifier-public-key-openshift-ci-2 >> "$$keydir/verifier-public-key-ci.gpg"; \
-	gpg --enarmor < "$$keydir/verifier-public-key-ci.gpg" > "$$keydir/verifier-public-key-ci"; \
-	sed -i 's/ARMORED FILE/PUBLIC KEY BLOCK/' "$$keydir/verifier-public-key-ci"; \
-	echo "# Release verification against OpenShift CI keys signed by the CI infrastructure" > \
-		manifests/0000_90_cluster-update-keys_configmap.yaml; \
-	oc create configmap release-verification \
-			--from-file=$$keydir/verifier-public-key-ci \
-			--from-file=stores/store-openshift-ci-release \
-			--dry-run -o yaml | \
-		oc annotate -f - release.openshift.io/verification-config-map= \
-			include.release.openshift.io/ibm-cloud-managed="true" \
-			include.release.openshift.io/self-managed-high-availability="true" \
-			include.release.openshift.io/single-node-developer="true" \
-			-n openshift-config-managed --local --dry-run=client -o yaml	>> \
-		manifests/0000_90_cluster-update-keys_configmap.yaml; \
-	echo "  namespace: openshift-config-managed" >> \
-		manifests/0000_90_cluster-update-keys_configmap.yaml
+	bash hack/generate-configmap.sh --ci
```

Comment:
Not a big deal, because these rebuilds are not expensive, but personally I prefer tracking dependencies in the `Makefile` (like [this][1], although even then, I had phony `ci`, etc., targets instead of using `manifests/0000_90_cluster-update-keys_configmap.yaml` and similar targets like I should have), instead of moving dependency/target information into the script itself.  I'm having trouble putting a finger on why.  I guess it's fewer places to read (just the `Makefile`) to understand how the various files are wired together.  But it obviously works the way you have it too, so :shrug:

[1]: https://github.com/openshift/cluster-update-keys/pull/16/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R7

Reply:
Yes, it’s so much better to at least track the dependencies in the Makefile for a better understanding of the files and how they are wired together.

I tried to make the dependencies clear in the Makefile in the new changes. But they are still hard-coded in the script as well. I could change the current directory tree structure and dynamically load every file based on the parent directory (see example). Or create a configuration file somehow. Or leave it be. What do you think?

For example:
.
├── hack
│   ├── generate-configmap.sh
│   └── verify-configmaps.sh
│
├── rhel
│   ├── keys
│   │   ├── verifier-public-key-redhat-release
│   │   └── verifier-public-key-redhat-beta-2
│   └── stores
│       ├── store-openshift-official-release
│       └── store-openshift-official-release-mirror
├── ci
│   ├── keys
│   │   ├── verifier-public-key-openshift-ci
│   │   └── verifier-public-key-openshift-ci-2
│   └── stores
│	   └── store-openshift-ci-release
│
├── manifests
│   └── 0000_90_cluster-update-keys_configmap.yaml
├── manifests.rhel
│   └── 0000_90_cluster-update-keys_configmap.yaml

Reply:
Yes, it’s so much better to at least track the dependencies in the `Makefile` for a better understanding of the files and how they are wired together.

I tried to make the dependencies more clear in the Makefile in the new commit https://github.com/openshift/cluster-update-keys/commit/d8197c0d0a485af7bb0b9f423e461f9606989991. But they are still hard-coded in the script as well. I could change the current directory tree structure and dynamically load every file based on the parent directory. Or create a configuration file somehow. Or leave it be. What do you think?